File: frontend/src/types/todo.ts
/**
 * 待办事项完整数据结构
 */
export interface Todo {
  /** 唯一标识符 */
  id: number
  /** 标题 */
  title: string
  /** 描述信息 */
  description: string | null
  /** 是否已完成 */
  completed: boolean
  /** 所有者ID */
  owner_id: number
  /** 创建时间 */
  created_at?: string
  /** 更新时间 */
  updated_at?: string
}

/**
 * 创建待办事项所需数据
 */
export interface TodoCreateData {
  /** 标题 */
  title: string
  /** 描述信息 */
  description?: string
  /** 是否已完成 */
  completed?: boolean
}

/**
 * 更新待办事项所需数据
 */
export interface TodoUpdateData {
  /** 标题 */
  title?: string
  /** 描述信息 */
  description?: string
  /** 是否已完成 */
  completed?: boolean
}

File: backend/app/models/todo.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from ..services.db import Base


class Todo(Base):
    __tablename__ = "todos"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=False, index=True)
    description = Column(String, nullable=True)
    completed = Column(Boolean, default=False)
    owner_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    owner = relationship("User", back_populates="todos") 